services/database/scripts.mjs

```diff
@@ -0,0 +1,329 @@
+
+    // 1. Read contracts from repository
+    try {
+      let contracts = [];
+
+      for await (const entry of readdirp(repositoryV1Path, {
+        alwaysStat: true,
+        depth: 2,
+        type: "directories",
+      })) {
+        const pathParts = entry.fullPath.split("/");
+        const address = pathParts.pop();
+        const chainId = pathParts.pop();
+        const matchType = pathParts.pop();
+
+        if (matchType === "full_match" || matchType === "partial_match") {
+          contracts.push({
+            chainId,
+            address,
+            timestamp: entry.stats.birthtime,
+            matchType,
+          });
+        }
+      }
+
+      contractsSorted = contracts.sort(
+        (o1, o2) => o1.timestamp.getTime() - o2.timestamp.getTime()
+      );
+
+      if (options.startFrom) {
+        contractsSorted = contractsSorted.filter(
+          (obj) => obj.timestamp.getTime() > options.startFrom
+        );
+      }
+
+      if (options.until) {
+        contractsSorted = contractsSorted.filter(
+          (obj) => obj.timestamp.getTime() < options.until
+        );
+      }
+    } catch (e) {
+      console.error(
+        "Error while reading and formatting files from the repository"
+      );
+      console.error(e);
+      process.exit(3);
+    }
+
+    // 2. Insert contracts to sourcify_sync table
+    try {
+      const databasePool = new Pool({
+        host: process.env.POSTGRES_HOST,
+        port: process.env.POSTGRES_PORT,
+        database: process.env.POSTGRES_DB,
+        user: process.env.POSTGRES_USER,
+        password: process.env.POSTGRES_PASSWORD,
+        max: 5,
+      });
+
+      for (const contract of contractsSorted) {
+        await databasePool.query(
+          `INSERT INTO sourcify_sync (chain_id, address, match_type, created_at) VALUES ($1, $2, $3, $4) ON CONFLICT (chain_id, address) DO NOTHING`,
+          [
+            contract.chainId,
+            contract.address,
+            contract.matchType,
+            contract.timestamp,
+          ]
+        );
+        console.log(
+          `Contract inserted to the sourcify_sync table: \n ${[
+            contract.chainId,
+            contract.address,
+            contract.matchType,
+            contract.timestamp,
+          ]}`
+        );
+      }
+    } catch (e) {
+      console.error("Error while storing contract in the database");
+      console.error(e);
+      process.exit(3);
+    }
+
+    console.log("successfuly imported contracts from" + repositoryV1Path);
+    databasePool.end();
+  });
+
+program
+  .command("sync")
+  .description("Verifies contracts in the sourcify_sync table")
+  .argument(
+    "<string>",
+    "Sourcify instance url: e.g. https://sourcify.dev/server"
+  )
+  .argument(
+    "<string>",
+    "Path to repository v1 contracts folder (e.g. /Users/user/sourcify/repository/contracts)"
+  )
+  .option(
+    "-c, --chains [items]",
+    "List of chains separated by comma (e.g. 1,5,...)"
```

Comment:
If we leave it blank, does it do all the chains?

Reply:
I'll fix this 👍